packages/core/src/routes/experience/classes/verifications/social-verification.ts

```diff
@@ -0,0 +1,183 @@
+   * Create the authorization URL for the social connector.
+   * Store the connector session result in the provider's interaction storage.
+   *
+   * @remarks
+   * Refers to the {@link createSocialAuthorizationUrl} method in the interaction/utils/social-verification.ts file.
+   * Currently, all the intermediate connector session results are stored in the provider's interactionDetails separately,
+   * apart from the new verification record.
+   * For compatibility reasons, we keep using the old {@link createSocialAuthorizationUrl} method here as a single source of truth.
+   * Especially for the SAML connectors,
+   * SAML ACS endpoint will find the connector session result by the jti and assign it to the interaction storage.
+   * We will need to update the SAML ACS endpoint before move the logic to this new SocialVerification class.
+   *
+   * TODO: Consider store the connector session result in the verification record directly.
+   * SAML ACS endpoint will find the verification record by the jti and assign the connector session result to the verification record.
+   */
+  async createAuthorizationUrl(
+    ctx: WithLogContext,
+    tenantContext: TenantContext,
+    { state, redirectUri }: SocialAuthorizationUrlPayload
+  ) {
+    return createSocialAuthorizationUrl(ctx, tenantContext, {
+      connectorId: this.connectorId,
+      state,
+      redirectUri,
+    });
+  }
+
+  /**
+   * Verify the social identity and store the social identity in the verification record.
+   *
+   * - Store the social identity in the verification record.
+   * - Find the user by the social identity and store the userId in the verification record if the user exists.
+   *
+   * @remarks
+   * Refer to the {@link verifySocialIdentity} method in the interaction/utils/social-verification.ts file.
+   * For compatibility reasons, we keep using the old {@link verifySocialIdentity} method here as a single source of truth.
+   * See the above {@link createAuthorizationUrl} method for more details.
+   *
+   * TODO: check the log event
+   */
+  async verify(ctx: WithLogContext, tenantContext: TenantContext, connectorData: JsonObject) {
+    const socialUserInfo = await verifySocialIdentity(
+      { connectorId: this.connectorId, connectorData },
+      ctx,
+      tenantContext
+    );
+
+    this.socialUserInfo = socialUserInfo;
+
+    const user = await this.findUserBySocialIdentity();
+    this.userId = user?.id;
+  }
+
+  async findUserBySocialIdentity(): Promise<User | undefined> {
+    const { socials } = this.libraries;
+    const {
+      users: { findUserByIdentity },
+    } = this.queries;
+
+    if (!this.socialUserInfo) {
+      return;
+    }
+
+    const {
+      metadata: { target },
+    } = await socials.getConnector(this.connectorId);
+
+    const user = await findUserByIdentity(target, this.socialUserInfo.id);
+
+    return user ?? undefined;
+  }
+
+  async findRelatedUserBySocialIdentity(): ReturnType<typeof socials.findSocialRelatedUser> {
```

Comment:
nit: add some comments to this method about the difference between this and `findUserBySocialIdentity`

Reply:
Added